src/network/NetworkController.ts

```diff
@@ -288,6 +299,35 @@ export class NetworkController extends BaseController<
     });
     this.refreshNetwork();
   }
+
+  getEIP1559Compatibility() {
+    const { properties = {} } = this.state;
+
+    if (!properties.isEIP1559Compatible) {
+      if (!this.ethQuery || !this.ethQuery.sendAsync) {
+        return Promise.resolve(true);
+      }
+      return new Promise((resolve, reject) => {
+        this.ethQuery.sendAsync(
+          { method: 'eth_getBlockByNumber', params: ['latest', false] },
+          (error: Error, block: Block) => {
+            if (error) {
+              reject(error);
+            } else {
+              const isEIP1559Compatible = typeof block.baseFee !== undefined;
```

Comment:
I think this should be a string `'undefined'`

Reply:
Also, from testing on Ropsten, I believe it should be `block.baseFeePerGas`

Reply:
done in 2bdcd7d and baseFeePerGas is what we used on extension's version of this code with success.